core/app.cpp

```diff
@@ -1042,27 +1052,68 @@ namespace MR
       // if necessary, also check for pre-existence of any output files with known paths
       //   (if the output is e.g. given as a prefix, the argument should be flagged as type_text())
       for (const auto& i : argument) {
-        if ((i.arg->type == ArgFileIn || i.arg->type == TracksIn) && !Path::exists (std::string(i)))
-          throw Exception ("required input file \"" + str(i) + "\" not found");
-        if (i.arg->type == ArgFileOut || i.arg->type == TracksOut)
-          check_overwrite (std::string(i));
-        if (i.arg->type == TracksIn && !Path::has_suffix (str(i), ".tck"))
-          throw Exception ("input file " + str(i) + " is not a valid track file");
-        if (i.arg->type == TracksOut && !Path::has_suffix (str(i), ".tck"))
-          throw Exception ("output track file (" + str(i) + ") must use the .tck suffix");
+        const std::string text = std::string (i);
+        if (i.arg->type == ArgFileIn || i.arg->type == TracksIn) {
+          if (!Path::exists (text))
+            throw Exception ("required input file \"" + text + "\" not found");
+          if (!Path::is_file (text))
+            throw Exception ("required input \"" + text + "\" is not a file");
+        }
+        if (i.arg->type == ArgDirectoryIn) {
+          if (!Path::exists (text))
+            throw Exception ("required input directory \"" + text + "\" not found");
+          if (!Path::is_dir (text))
+            throw Exception ("required input \"" + text + "\" is not a directory");
+        }
+        if (i.arg->type == ArgFileOut || i.arg->type == TracksOut) {
+          if (text.find_last_of (PATH_SEPARATOR) == text.size() - std::string(PATH_SEPARATOR).size())
+            throw Exception ("output path \"" + std::string(i) + "\" is not a valid file path (ends with \'" PATH_SEPARATOR "\")");
+          check_overwrite (text);
+        }
+        if (i.arg->type == ArgDirectoryOut)
+          check_overwrite (text);
+        if (i.arg->type == TracksIn && !Path::has_suffix (text, ".tck"))
+          throw Exception ("input file \"" + text + "\" is not a valid track file");
+        if (i.arg->type == TracksOut && !Path::has_suffix (text, ".tck"))
+          throw Exception ("output track file \"" + text + "\" must use the .tck suffix");
+        if (i.arg->type == ArgDirectoryOut) {
+          const std::string base_name = Path::basename (text);
+          if (base_name.find (".") != std::string::npos)
+            throw Exception ("output path \"" + text + "\" is not a valid directory path (contains a \'.\')");
```

Comment:
Why explicitly disallow dots in the directory name? There's no particular reason Unix-wise to prevent this... (?)

Reply:
Yeah, I umm-ed and arr-ed about that for a while. Even though it's legal, it's not something I would generally recommend, and could see it leading to confusion in cases where the user does not fully understand that the output is intended to be a directory rather than a file (e.g. ending up with fixel directories with ".mif" postfixes). Happy for that test to be removed if you think it's too un-Unix-ey.

Reply:
I know what you mean. But while it would be odd to have directory with a `.mif` suffix, it wouldn't actually cause any problems for the code in any way - everything would work as expected. And preventing it would limit what users can legitimatelly do, e.g. they couldn't label their fixel folder `FBA_03.01.2018/`, which strikes me a perfectly sensible thing to call it. I'd rather take than out, I think that's trying too hard to guide novice users, when the worst case consequence of them mislabeling their folder is minor confusion...